bundles/org.openhab.binding.bsblan/src/main/java/org/openhab/binding/bsblan/internal/handler/BsbLanParameterHandler.java

```diff
@@ -0,0 +1,169 @@
+
+        // it will take up to refreshInterval seconds until we receive a value and thing goes online
+        // see notes in {@link BsbLanBridgeHandler#registerThing(BsbLanBaseThingHandler)}
+        updateStatus(ThingStatus.UNKNOWN);
+    }
+
+    /**
+     * Update the channel from the last data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    @Override
+    protected void updateChannel(String channelId) {
+        BsbLanApiParameterQueryResponseDTO data = null;
+        BsbLanBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            data = bridgeHandler.getCachedParameterQueryResponse();
+        }
+        updateChannel(channelId, data);
+    }
+
+    private void updateChannel(String channelId, @Nullable BsbLanApiParameterQueryResponseDTO data) {
+        if (data == null) {
+            logger.debug("no data available while updating channel '{}' of parameter {}", channelId, parameterConfig.id);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.BRIDGE_OFFLINE,
+                "No data received from BSB-LAN device");
+            return;
+        }
+
+        BsbLanApiParameterDTO parameter = data.getOrDefault(parameterConfig.id, null);
+        if (parameter == null) {
+            logger.debug("parameter {} is not part of response data while updating channel '{}' ", parameterConfig.id, channelId);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                String.format("No data received for parameter %s", parameterConfig.id));
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+
+        if (!isLinked(channelId)) {
+            return;
+        }
+
+        State state = BsbLanParameterConverter.getState(channelId, parameter);
+        if (state == null) {
+            return;
+        }
+
+        updateState(channelId, state);
+    }
+
+     /**
+     * Update the channel from the last data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     * @param command the value to be set
+     */
+    @Override
+    protected void setChannel(String channelId, Command command) {
+        logger.debug("Received command '{}' for channel '{}'", command, channelId);
+
+        if (!WRITEABLE_CHANNELS.contains(channelId)) {
+            logger.debug("Channel '{}' is read only. Ignoring command", channelId);
+            return;
+        }
+
+        String value = BsbLanParameterConverter.getValue(channelId, command);
+        if (value == null) {
+            logger.debug("Channel '{}' is read only or conversion failed. Ignoring command", channelId);
+            return;
+        }
+
+        BsbLanApiCaller api = getApiCaller();
+        if (api == null) {
+            logger.error("Failed to set parameter {} (API unavailable)", parameterConfig.setId);

```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. In error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

My general rule for logging levels is:
* trace - used for normal expected execution paths
* debug - used for unexepected execution paths, but well within normal operation
* info - used for notable points in an execution path, like a milestone. (in openhab we try reserve this logging level for the core, so bindings should rarely ever call this.)
* warn - used for notable unexpected execution paths that a regular user (not just a developer) should be notified of. Warnings should be used to indicate that something not-normal occurred and user intervention is required to resolve. Warnings do not indicate a failure to operate merely an abnormal condition of operation that can still be handled by the binding. Failures in binding operation should be indicated by changing the thing status to offline.
* error - used to indicate catastrophic program failure. This should be used to indicate a catastrophic failure in openhab's ability to operate. A failure in a binding would never cause openhab as a whole to fail so a failure in a bindings should never log an error. Instead that failure should be indicated by changing the thing status.

Reply:
Changed the log levels, think they should now be as expected in the guidelines.